components/authz-service/server/v2/project_test.go

```diff
@@ -336,6 +336,54 @@ func TestListProjects(t *testing.T) {
 			assert.ElementsMatch(t, []*api.Project{&apiProject0, &apiProject1}, resp.Projects)
 		}},
 	}
+	rand.Shuffle(len(cases), func(i, j int) {
+		cases[i], cases[j] = cases[j], cases[i]
+	})
+
+	for _, test := range cases {
+		t.Run(test.desc, test.f)
+		store.Flush()
+	}
+}
+func TestListProjectsForIntrospection(t *testing.T) {
+	ctx := context.Background()
+	cl, store, _ := setupProjects(t)
+
+	cases := []struct {
+		desc string
+		f    func(*testing.T)
+	}{
+		{"if no projects exist, returns empty list", func(t *testing.T) {
+			resp, err := cl.ListProjectsForIntrospection(ctx, &api.ListProjectsReq{})
+			require.NoError(t, err)
+			assert.Empty(t, resp.Projects)
+		}},
+		{"if two projects exist, returns these projects", func(t *testing.T) {
+			require.Zero(t, store.ItemCount())
+			id0 := "foo-project"
+			apiProject0 := addProjectToStore(t, store, id0, "my foo", storage.ChefManaged)
+
+			id1 := "test-2-project"
+			apiProject1 := addProjectToStore(t, store, id1, "my bar", storage.Custom)
+
+			resp, err := cl.ListProjectsForIntrospection(ctx, &api.ListProjectsReq{})
+			require.NoError(t, err)
+			assert.ElementsMatch(t, []*api.Project{&apiProject0, &apiProject1}, resp.Projects)
+		}},
+		{"suppresses hidden system projects except for unassigned", func(t *testing.T) {
+			require.Zero(t, store.ItemCount())
+			addProjectToStore(t, store, constants.AllProjectsID, "All Projects", storage.ChefManaged)
+			unassignedProject := addProjectToStore(t, store, constants.UnassignedProjectID, "Unassigned", storage.ChefManaged)
```

Comment:
Seems like these projects should *always* exist in the store (rather than be added for this test) as that is more comparable to how our system is in production/